### PR TITLE
feat(pt): Support denoise model

### LINKED ISSUE #3493: [Feature Request] pt: support denoise in the PyTorch backend

### PR DESCRIPTION
**Pull request by Chengqian-Zhang**
**Training process**:
Add token, coordinate and cell noise to the equilibrium structure. Then we use model to predict these noises.
**Application**:
- Relax near-equilibrium structures. It is useful to improve the accuracy of IS2RP(P: Property) task according to @zjgemi's  experiments in HEA datasets using code in this PR.
- Pretraining (model parameter warm up). 

## Summary by CodeRabbit

- **New Features**
  - Integrated advanced denoising capabilities with flexible noise handling and masking strategies.
  - Introduced a specialized atomic model (`DPDenoiseAtomicModel`) and a dedicated denoising model (`DenoiseModel`) for improved predictive performance.
  - Added a new configuration file (`input.json`) for denoising model training with detailed parameters.
  - Created new data files containing chemical element symbols for denoising examples.
  - Added new files for denoising data sequences (`type.raw` and `type_map.raw`) in multiple directories.
  - Introduced a new loss type (`DenoiseLoss`) for enhanced denoising functionality.

- **Enhancements**
  - Overhauled loss computation and fitting routines to offer refined control over noise processing.
  - Expanded configuration options and added robust error handling to ensure proper usage.

- **Utilities**
  - Added new argument validation functions (`fitting_denoise` and `loss_denoise`) to streamline the setup of denoising tasks.

- **Bug Fixes**
  - Enhanced output structure in evaluation functions to include strain components and updated coordinates during model evaluation.

- **Tests**
  - Introduced new test classes and modified existing ones to validate denoising functionalities across different model configurations.
  - Removed outdated test classes that were not compatible with the current denoising implementation.
  - Enhanced test coverage for the `DenoiseModel` and `DenoiseFittingNet` functionalities.

### CODE REVIEW

File: deepmd/pt/model/task/denoise.py [hunk 1]
```diff
@@ -83,55 +307,309 @@
...
+            # "spin": self.spin ,
+            ## NOTICE:  not supported by far
+            "tot_ener_zero": False,
+            "trainable": [self.trainable] * (len(self.neuron) + 1),
+            "layer_name": None,
+            "use_aparam_as_mask": self.use_aparam_as_mask,
+            "spin": None,
+        }
+
+    def deserialize(cls, data: dict) -> "DenoiseNet":
```
[the remark this answers is not in the document]
Comment (Chengqian-Zhang) [reply deepmd/pt/model/task/denoise.py:362]: Solved

File: source/tests/pt/common.py [hunk 2]
```diff
@@ -210,8 +226,20 @@
             if atomic_virial_out
             else np.zeros([nframes, natoms, 3, 3])  # pylint: disable=no-explicit-dtype
         )
+        strain_components_out = (
+            np.concatenate(strain_components_out)
+            if strain_components_out
+            else np.zeros([nframes, 6])  # pylint: disable=no-explicit-dtype
```
[the remark this answers is not in the document]
Comment (Chengqian-Zhang) [reply source/tests/pt/common.py:232]: solved

File: source/tests/pt/common.py [hunk 3]
```diff
@@ -257,10 +285,35 @@
                 [nframes, natoms, 3, 3], dtype=GLOBAL_PT_FLOAT_PRECISION, device=DEVICE
             )
         )
-        updated_coord_out = torch.cat(updated_coord_out) if updated_coord_out else None
+        strain_components_out = (
+            torch.cat(strain_components_out)
+            if strain_components_out
+            else torch.zeros(
```
[the remark this answers is not in the document]
Comment (Chengqian-Zhang) [reply source/tests/pt/common.py:291]: solved

File: source/tests/pt/common.py [hunk 4]
```diff
@@ -257,10 +285,35 @@
...
+                [nframes, 6], dtype=GLOBAL_PT_FLOAT_PRECISION, device=DEVICE
+            )
+        )
+        atom_strain_components_out = (
```
[the remark this answers is not in the document]
Comment (Chengqian-Zhang) [reply source/tests/pt/common.py:295]: solved

File: source/tests/universal/pt/model/test_model.py [hunk 5]
```diff
@@ -919,3 +925,95 @@
...
+        return modules
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        DenoiseModelTest.setUpClass()
+        (DescriptorParam, Descrpt) = cls.param[0]
+        (FittingParam, Fitting) = cls.param[1]
+        # set special precision
+        # if Descrpt in [DescrptDPA2]:
+        #    cls.epsilon_dict["test_smooth"] = 1e-8
```
[the remark this answers is not in the document]
Comment (Chengqian-Zhang) [reply source/tests/universal/pt/model/test_model.py:976]: solved

**Comment by njzjz** [comment]
Possible reviewers based on changed lines, exact file history, and exact-file review history:

- @iProzd — 118 commits on changed files (`deepmd/dpmodel/model/__init__.py`, `deepmd/dpmodel/model/model.py`, `deepmd/entrypoints/test.py`, `deepmd/pt/infer/deep_eval.py`, `deepmd/pt/loss/__init__.py`, `deepmd/pt/loss/denoise.py`, `deepmd/pt/model/model/__init__.py`, `deepmd/pt/model/task/__init__.py`, `deepmd/pt/model/task/denoise.py`, `deepmd/pt/train/training.py`, `deepmd/pt/utils/stat.py`, `deepmd/utils/argcheck.py`, `source/tests/pt/model/test_permutation.py`, `source/tests/pt/model/test_permutation_denoise.py`, `source/tests/pt/model/test_rot.py`, `source/tests/pt/model/test_rot_denoise.py`, `source/tests/pt/model/test_smooth.py`, `source/tests/pt/model/test_smooth_denoise.py`, `source/tests/pt/model/test_trans.py`, `source/tests/pt/model/test_trans_denoise.py`, `source/tests/universal/common/cases/atomic_model/atomic_model.py`, `source/tests/universal/common/cases/fitting/utils.py`, `source/tests/universal/common/cases/loss/utils.py`, `source/tests/universal/common/cases/model/model.py`, `source/tests/universal/dpmodel/atomc_model/test_atomic_model.py`, `source/tests/universal/dpmodel/fitting/test_fitting.py`, `source/tests/universal/dpmodel/loss/test_loss.py`, `source/tests/universal/pt/atomc_model/test_atomic_model.py`, `source/tests/universal/pt/fitting/test_fitting.py`, `source/tests/universal/pt/loss/test_loss.py`, `source/tests/universal/pt/model/test_model.py`).
- @wanghan-iapcm — 45 commits on changed files; 45 reviews on exact changed files (`deepmd/dpmodel/atomic_model/__init__.py`, `deepmd/dpmodel/fitting/__init__.py`, `deepmd/dpmodel/model/__init__.py`, `deepmd/dpmodel/model/model.py`, `deepmd/entrypoints/test.py`, `deepmd/pt/infer/deep_eval.py`, `deepmd/pt/model/atomic_model/__init__.py`, `deepmd/pt/model/model/__init__.py`, `deepmd/pt/model/task/__init__.py`, `deepmd/pt/model/task/denoise.py`, `deepmd/pt/train/training.py`, `deepmd/pt/utils/stat.py`, `deepmd/utils/argcheck.py`, `source/tests/pt/model/test_permutation.py`, `source/tests/pt/model/test_permutation_denoise.py`, `source/tests/pt/model/test_rot.py`, `source/tests/pt/model/test_rot_denoise.py`, `source/tests/pt/model/test_smooth.py`, `source/tests/pt/model/test_smooth_denoise.py`, `source/tests/pt/model/test_trans.py`, `source/tests/pt/model/test_trans_denoise.py`).

No review request was made automatically.

Coding agent: Codex
Codex version: codex-cli 0.144.4
Model: gpt-5.6-sol
Reasoning effort: xhigh
